### ISSUE TITLE
Crash on Pairing

### ISSUE THREAD

**Issue by AzonInc** [uriyacovy/ESPHome_nuki_lock#40]
Whenever I'm trying to pair the smart lock the ESP is crashing and goes into a bootloop.
I have the Smartlock 4.0 Pro with Wifi disabled (2m away from the lock).

Might be related to #24 
But it works on nuki_hub.

Hardware: NodeMCU ESP32

Config:
```
substitutions:
  name: tcs-intercom-test
  friendly_name: TCS Intercom Test

api:

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

ota:

logger:
  level: DEBUG

web_server:

esphome:
  name: ${name}
  name_add_mac_suffix: false
  friendly_name: ${friendly_name}
  libraries:
  - Preferences
  - https://github.com/vinmenn/Crc16.git
  - https://github.com/uriyacovy/NukiBleEsp32

external_components:
  - source: github://uriyacovy/ESPHome_nuki_lock@dev
  
esp32:
  board: "nodemcu-32s"  # Or whatever other board you're using
  framework:
    type: arduino
    version: 2.0.3
    platform_version: 5.3.0

lock:
  # Required:
  - platform: nuki_lock
    name: Nuki Lock
    is_connected: 
      name: "Nuki Connected"
    is_paired: 
      name: "Nuki Paired"

  # Optional:
    battery_critical:
      name: "Nuki Battery Critical"
    battery_level:
      name: "Nuki Battery Level"
    door_sensor:
      name: "Nuki Door Sensor"
    door_sensor_state:
      name: "Nuki Door Sensor State"

  #  unpair: True
```

Logs:
```
[ 45805][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[ 45806][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[ 46305][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[ 46305][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[ 46805][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[ 46805][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[ 47305][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[ 47305][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[ 47805][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[ 47805][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[ 48306][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[ 48306][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[ 48805][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[ 48805][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[ 49305][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[ 49305][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[ 49805][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[ 49805][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[ 50305][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[ 50305][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[ 50805][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[ 50805][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[ 51305][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[ 51305][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue






############################################
########## Pressed Button on Nuki to pair ##########
############################################






E (59211) task_wdt: Task watchdog got triggered. The following tasks did not reset the watchdog in time:
E (59211) task_wdt:  - loopTask (CPU 1)
E (59211) task_wdt: Tasks currently running:
E (59211) task_wdt: CPU 0: IDLE
E (59211) task_wdt: CPU 1: IDLE
E (59211) task_wdt: Aborting.

abort() was called at PC 0x40102964 on core 0


Backtrace:0x40083709:0x3ffbea0c |<-CORRUPTED




ELF file SHA256: 0000000000000000

Rebooting...
ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:12784
load:0x40080400,len:3032
entry 0x400805e4
[I][logger:158]: Log initialized
[C][ota:483]: There have been 2 suspected unsuccessful boot attempts.
[D][esp32.preferences:114]: Saving 1 preferences to flash...
[D][esp32.preferences:143]: Saving 1 preferences to flash: 0 cached, 1 written, 0 failed
[I][app:029]: Running through setup()...
[I][nukilock.lock:160]: Starting NUKI Lock...
[   345][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[   346][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[W][nukilock.lock:183]: l��? Nuki is not paired
[D][binary_sensor:034]: 'Nuki Paired': Sending initial state OFF
[D][lock:054]: 'Nuki Lock': Sending state UNKNOWN
[C][wifi:038]: Setting up WiFi...
[C][wifi:051]: Starting WiFi...
[C][wifi:052]:   Local MAC: 0C:B8:15:F8:18:B0
[D][wifi:462]: Starting scan...
[   437][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[   438][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[W][wifi:152]: Warning set: unspecified
[   725][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[   725][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[  1225][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[  1225][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[  1725][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[  1725][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
lld_pdu_get_tx_flush_nb HCI packet count mismatch (1, 2)
[W][component:232]: Component nuki_lock.lock took a long time for an operation (508 ms).
[W][component:233]: Components should block for at most 30 ms.
[  2250][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[  2250][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
lld_pdu_get_tx_flush_nb HCI packet count mismatch (1, 2)
[W][component:232]: Component nuki_lock.lock took a long time for an operation (429 ms).
[W][component:233]: Components should block for at most 30 ms.
[  2728][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[  2728][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
lld_pdu_get_tx_flush_nb HCI packet count mismatch (1, 2)
[W][component:232]: Component nuki_lock.lock took a long time for an operation (1458 ms).
[W][component:233]: Components should block for at most 30 ms.
[D][wifi:477]: Found networks:
[I][wifi:521]: - 'AZ-Wlan' [redacted]▂▄▆█
[D][wifi:522]:     Channel: 6
[D][wifi:523]:     RSSI: -55 dB
[D][wifi:526]: - [redacted] [redacted]▂▄▆█
[D][wifi:526]: - [redacted] [redacted]▂▄▆█
[D][wifi:526]: - [redacted] [redacted]▂▄▆█
[D][wifi:526]: - [redacted] [redacted]▂▄▆█
[D][wifi:526]: - [redacted] [redacted]▂▄▆█
[D][wifi:526]: - [redacted] [redacted]▂▄▆█
[D][wifi:526]: - [redacted] [redacted]▂▄▆█
[D][wifi:526]: - [redacted] [redacted]▂▄▆█
[D][wifi:526]: - [redacted] [redacted]▂▄▆█
[I][wifi:303]: WiFi Connecting to 'AZ-Wlan'...
[  4318][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[  4319][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
E (18913) task_wdt: Task watchdog got triggered. The following tasks did not reset the watchdog in time:
E (18913) task_wdt:  - loopTask (CPU 1)
E (18913) task_wdt: Tasks currently running:
E (18913) task_wdt: CPU 0: IDLE
E (18913) task_wdt: CPU 1: IDLE
E (18913) task_wdt: Aborting.

abort() was called at PC 0x40102964 on core 0


Backtrace:0x40083709:0x3ffbea0c |<-CORRUPTED




ELF file SHA256: 0000000000000000

Rebooting...
ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:12784
load:0x40080400,len:3032
entry 0x400805e4
[I][logger:158]: Log initialized
[C][ota:483]: There have been 3 suspected unsuccessful boot attempts.
[D][esp32.preferences:114]: Saving 1 preferences to flash...
[D][esp32.preferences:143]: Saving 1 preferences to flash: 0 cached, 1 written, 0 failed
[I][app:029]: Running through setup()...
[I][nukilock.lock:160]: Starting NUKI Lock...
[   351][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[   352][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[W][nukilock.lock:183]: l��? Nuki is not paired
[D][binary_sensor:034]: 'Nuki Paired': Sending initial state OFF
[D][lock:054]: 'Nuki Lock': Sending state UNKNOWN
[C][wifi:038]: Setting up WiFi...
[C][wifi:051]: Starting WiFi...
[C][wifi:052]:   Local MAC: 0C:B8:15:F8:18:B0
[D][wifi:462]: Starting scan...
[   442][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[   443][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[W][wifi:152]: Warning set: unspecified
[   693][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[   693][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
lld_pdu_get_tx_flush_nb HCI packet count mismatch (1, 2)
[W][component:232]: Component nuki_lock.lock took a long time for an operation (797 ms).
[W][component:233]: Components should block for at most 30 ms.
[  1505][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[  1505][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
lld_pdu_get_tx_flush_nb HCI packet count mismatch (1, 2)
[W][component:232]: Component nuki_lock.lock took a long time for an operation (858 ms).
[W][component:233]: Components should block for at most 30 ms.
[  2374][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[  2374][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
lld_pdu_get_tx_flush_nb HCI packet count mismatch (1, 2)
[W][component:232]: Component nuki_lock.lock took a long time for an operation (420 ms).
[W][component:233]: Components should block for at most 30 ms.
[D][wifi:477]: Found networks:
[I][wifi:521]: - 'AZ-Wlan' [redacted]▂▄▆█
[D][wifi:522]:     Channel: 6
[D][wifi:523]:     RSSI: -56 dB
[D][wifi:526]: - [redacted] [redacted]▂▄▆█
[D][wifi:526]: - [redacted] [redacted]▂▄▆█
[D][wifi:526]: - [redacted] [redacted]▂▄▆█
[D][wifi:526]: - [redacted] [redacted]▂▄▆█
[D][wifi:526]: - [redacted] [redacted]▂▄▆█
[D][wifi:526]: - [redacted] [redacted]▂▄▆█
[I][wifi:303]: WiFi Connecting to 'AZ-Wlan'...
[  2893][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[  2894][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
lld_pdu_get_tx_flush_nb HCI packet count mismatch (1, 2)
[W][component:232]: Component nuki_lock.lock took a long time for an operation (771 ms).
[W][component:233]: Components should block for at most 30 ms.
[I][wifi:594]: WiFi Connected!
[C][wifi:408]:   Local MAC: 0C:B8:15:F8:18:B0
[C][wifi:413]:   SSID: [redacted]
[C][wifi:416]:   IP Address: 10.10.0.109
[C][wifi:420]:   BSSID: [redacted]
[C][wifi:421]:   Hostname: 'tcs-intercom-test'
[C][wifi:423]:   Signal strength: -56 dB ▂▄▆█
[C][wifi:427]:   Channel: 6
[C][wifi:428]:   Subnet: 255.255.0.0
[C][wifi:429]:   Gateway: 10.10.0.1
[C][wifi:430]:   DNS1: 10.10.0.2
[C][wifi:431]:   DNS2: 0.0.0.0
[C][web_server:124]: Setting up web server...
[C][ota:096]: Over-The-Air Updates:
[C][ota:097]:   Address: tcs-intercom-test.local:3232
[C][ota:103]:   OTA version: 2.
[W][ota:107]: Last Boot was an unhandled reset, will proceed to safe mode in 7 restarts
[C][api:025]: Setting up Home Assistant API server...
[I][app:062]: setup() finished successfully!
[  3775][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[  3776][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
lld_pdu_get_tx_flush_nb HCI packet count mismatch (1, 2)
[W][component:232]: Component nuki_lock.lock took a long time for an operation (970 ms).
[W][component:233]: Components should block for at most 30 ms.
[W][wifi:165]: Warning cleared
[W][api:152]: Warning set: unspecified
[I][app:102]: ESPHome version 2024.3.0 compiled on Mar 25 2024, 14:29:26
[C][wifi:580]: WiFi:
[C][wifi:408]:   Local MAC: 0C:B8:15:F8:18:B0
[C][wifi:413]:   SSID: [redacted]
[C][wifi:416]:   IP Address: 10.10.0.109
[C][wifi:420]:   BSSID: [redacted]
[C][wifi:421]:   Hostname: 'tcs-intercom-test'
[C][wifi:423]:   Signal strength: -56 dB ▂▄▆█
[C][wifi:427]:   Channel: 6
[C][wifi:428]:   Subnet: 255.255.0.0
[C][wifi:429]:   Gateway: 10.10.0.1
[C][wifi:430]:   DNS1: 10.10.0.2
[C][wifi:431]:   DNS2: 0.0.0.0
[  4833][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[  4834][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
lld_pdu_get_tx_flush_nb HCI packet count mismatch (1, 2)
[W][component:232]: Component nuki_lock.lock took a long time for an operation (342 ms).
[W][component:233]: Components should block for at most 30 ms.
[C][logger:166]: Logger:
[C][logger:167]:   Level: DEBUG
[C][logger:169]:   Log Baud Rate: 115200
[C][logger:170]:   Hardware UART: UART0
[  5211][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[  5212][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
E (20698) task_wdt: Task watchdog got triggered. The following tasks did not reset the watchdog in time:
E (20698) task_wdt:  - loopTask (CPU 1)
E (20698) task_wdt: Tasks currently running:
E (20698) task_wdt: CPU 0: IDLE
E (20698) task_wdt: CPU 1: IDLE
E (20698) task_wdt: Aborting.

abort() was called at PC 0x40102964 on core 0


Backtrace:0x40083709:0x3ffbea0c |<-CORRUPTED




ELF file SHA256: 0000000000000000

Rebooting...
ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:12784
load:0x40080400,len:3032
entry 0x400805e4
[I][logger:158]: Log initialized
[C][ota:483]: There have been 4 suspected unsuccessful boot attempts.
[D][esp32.preferences:114]: Saving 1 preferences to flash...
[D][esp32.preferences:143]: Saving 1 preferences to flash: 0 cached, 1 written, 0 failed
[I][app:029]: Running through setup()...
[I][nukilock.lock:160]: Starting NUKI Lock...
[   348][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[   349][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[W][nukilock.lock:183]: l��? Nuki is not paired
[D][binary_sensor:034]: 'Nuki Paired': Sending initial state OFF
[D][lock:054]: 'Nuki Lock': Sending state UNKNOWN
[C][wifi:038]: Setting up WiFi...
[C][wifi:051]: Starting WiFi...
[C][wifi:052]:   Local MAC: 0C:B8:15:F8:18:B0
[D][wifi:462]: Starting scan...
[   440][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[   441][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[W][wifi:152]: Warning set: unspecified
[   717][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[   717][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[  1217][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[  1217][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
[  1717][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: bleAddress NOT_FOUND
[  1717][E][NukiBle.cpp:635] retrieveCredentials(): Getting data from NVS issue
E (13717) task_wdt: Task watchdog got triggered. The following tasks did not reset the watchdog in time:
E (13717) task_wdt:  - loopTask (CPU 1)
E (13717) task_wdt: Tasks currently running:
E (13717) task_wdt: CPU 0: IDLE
E (13717) task_wdt: CPU 1: IDLE
E (13717) task_wdt: Aborting.

abort() was called at PC 0x40102964 on core 0


Backtrace:0x40083709:0x3ffbea0c |<-CORRUPTED




ELF file SHA256: 0000000000000000

Rebooting...
ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:12784
load:0x40080400,len:3032
entry 0x400805e4
```

**Comment by JakobTewes** [comment]
Heyhey @AzonInc,

as I don´t even get it to pairing...maybe you can help me ;-). How did you pair the NUKI?
I tried it with "press-button-on-Nuki-until-ring-lights-up" and reboot the ESP32.
The NUKI had a pin set in the beginning, then i reset it and did not set a pin (but set it up via app).

Thanks and regards

Jakob

**Comment by AzonInc** [comment]
> Heyhey @AzonInc,
> 
> as I don´t even get it to pairing...maybe you can help me ;-). How did you pair the NUKI? I tried it with "press-button-on-Nuki-until-ring-lights-up" and reboot the ESP32. The NUKI had a pin set in the beginning, then i reset it and did not set a pin (but set it up via app).
> 
> Thanks and regards
> 
> Jakob

Actually I tried it several times until it "paired" and failed :(
I did it like u said - just pressing the button on the nuki for a few seconds until it lights up and then it immediately tried to pair and crashed right afterwards into a bootloop.

I'm using my fork of the project with a few additions to be able to initiate pairing mode myself so it doesn't try to pair all the time and only when neccessary.
However it doesnt make any difference while using the dev branch here also fails.

Maybe it's somewhat related to the 4th generation Locks.

Did you try nuki_hub already? That worked for me, even tho the connection was terrible.
